### PR TITLE
Add swagger-restify-mw

### PR DESCRIPTION
**Pull request by micmro**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `npm run new-package package-name`, not by basing it on an existing project.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: add explicit typings for `restify` (see [source](https://github.com/theganyo/swagger-node-runner/blob/master/lib/restify_middleware.js)) and tests to `swagger-node-runner`.
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by micmro** [comment]
👍

File: swagger-node-runner/swagger-node-runner-tests.ts [hunk 1]
```diff
@@ -45,6 +46,22 @@ SwaggerNodeRunner.create(config, function(err, runner) {
 });
 
 
+// Restify Middelware
+
+let app = restify.createServer();
+SwaggerNodeRunner.create(config, function(err, runner) {
+// SwaggerRestify.create(config, function(err, swaggerRestify) {
```
Comment (DanielRosenwasser) [inline swagger-node-runner/swagger-node-runner-tests.ts]: Can you remove this line?

Comment (micmro) [inline swagger-node-runner/swagger-node-runner-tests.ts]: Oups - removed now.

**Comment by DanielRosenwasser** [comment]
Is there any specific reason that the middleware needs to be exposed from `swagger-node-runner` rather than `swagger-node-mw` itself? Is it simply a convenience thing?

**Comment by micmro** [comment]
Yes `swagger-restify-mw` is just a light convenience wrapper around `swagger-node-runner`.
You can check the (very small) source here: https://github.com/apigee-127/swagger-restify/blob/master/lib/index.js

File: swagger-restify-mw/swagger-restify-mw-tests.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  if (err) { throw err; }
+
+  swaggerRestify.register(app);
+
+  let port = process.env.PORT || 10010;
+  app.listen(port);
+
+  // if (swaggerRestify.runner.swagger.paths['/hello']) {
+  //   console.log('try this:\ncurl http://127.0.0.1:' + port + '/hello?name=Scott');
+  // }
```
Comment (DanielRosenwasser) [inline swagger-restify-mw/swagger-restify-mw-tests.ts]: Can we also just get rid of this line as well since it's not being used?

**Comment by DanielRosenwasser** [comment]
Thanks!

**Comment by DanielRosenwasser** [comment]
To answer your question - no I don't think re-exporting is a problem. Don't sweat it.

**Comment by micmro** [comment]
Great, thanks! And happy new year!

**Pull request merged**